heat/fft/fft.py

```diff
@@ -0,0 +1,787 @@
+"""Provides a collection of Discrete Fast Fourier Transforms (DFFT) and their inverses."""
+
+import torch
+
+from ..core.communication import MPI
+from ..core.dndarray import DNDarray
+from ..core.stride_tricks import sanitize_axis
+from ..core.types import promote_types, heat_type_of
+from ..core.factories import array, zeros
+
+from typing import Type, Union, Tuple, Any, Iterable, Optional
+
+__all__ = [
+    "fft",
+    "fft2",
+    "fftn",
+    "hfft",
+    "hfft2",
+    "hfftn",
+    "ifft",
+    "ifft2",
+    "ifftn",
+    "ihfft",
+    "ihfft2",
+    "ihfftn",
+    "irfft",
+    "irfft2",
+    "irfftn",
+    "rfft",
+    "rfft2",
+    "rfftn",
+    # "fftfreq",
```

Comment:
If there is no special reason for omitting these functions, I would suggest to open a separate issue to add them lateron.

Reply:
these are implemented now